app/commands/mentor/discussion/finish_by_student.rb

```diff
@@ -90,6 +91,10 @@ def award_badge!
         AwardBadgeJob.perform_later(discussion.mentor, :mentor)
       end
 
+      def update_roles!
+        Mentor::UpdateRoles.defer(discussion.mentor)
```

Comment:
Nice.

Reply:
I feel like maybe I'd rather this was `User::UpdateMentorRoles`. I would have found that naming more obvious.